bin/oe/run.rs

```diff
@@ -450,7 +450,13 @@ pub fn execute(cmd: RunCmd, logger: Arc<RotatingLogger>) -> Result<RunningClient
     let is_ready = Arc::new(atomic::AtomicBool::new(true));
     miner.add_transactions_listener(Box::new(move |_hashes| {
         // we want to have only one PendingTransactions task in the queue.
-        if is_ready.compare_and_swap(true, false, atomic::Ordering::SeqCst) {
+        if is_ready.compare_exchange(
+            true,
+            false,
+            atomic::Ordering::SeqCst,
+            atomic::Ordering::SeqCst,
+        ) == Ok(true)
```

Comment:
I little bit dislike that we have `true` in `Ok(true)` it doesn't feel nice to have it in two places, especially if it could be changed to become:
`if let Ok(_) = is_ready.compare_exhange(....) {`
Can you change all others in same way?

Reply:
I missed this.

What about  ` if is_ready.compare_exhange(....).is_ok()` ?

Reply:
even better :D

Reply:
Switched to `is_ok`/`is_err`, is it also ok to remove a temporary `prev_last_sealed_block` (which I did)?